trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/FunctionHandler.java

```diff
@@ -396,17 +400,20 @@ public Result handleFunctionDefinition(final IDispatcher main, final MemoryHandl
 		return new Result(impl);
 	}
 
-	private static CFunction updateVarArgsUsage(final IASTFunctionDefinition node, final CFunction oldFunType) {
-		final CFunction funType;
+	private static CFunction updateVarArgsUsage(final ILocation loc, final IASTFunctionDefinition node,
+			final CFunction funType, final String funName) {
 		// update varags usage
-		if (oldFunType.hasVarArgs() && oldFunType.getVarArgsUsage() == VarArgsUsage.UNKNOWN) {
+		if (funType.hasVarArgs() && funType.getVarArgsUsage() == VarArgsUsage.UNKNOWN) {
 			// if the function body creates a va_list object it uses its varargs
-			final ASTNameCollector vaListFinder = new ASTNameCollector("va_list");
+			final ASTNameCollector vaListFinder = new ASTNameCollector("__builtin_va_start");

```

Comment:
would it make sense to check against both `va_start` and `__builtin_va_start` ?

I guess my general point is, the `__builtin_va_*`-Macros depend on a particular preprocessing -- can we also support files without that preprocessing?

Reply:
I see `StandardFunctionHandler` already considers this case.

Reply:
I just updated it. However to support varargs for files without preprocessing, we need also macros (in `GccStaticLanguageSettingsProvider`) for `va_list` and `va_arg`. Should I also add them or should we leave that for now?

Reply:
Is there a reason against it? If not, I'd suggest so -- it's better than leaving it half-implemented.

That brings up another question I had: You say we need a macro for `va_list`, but I don't see one for `__builtin_va_list`. How is that handled?

Reply:
Sure, I can just add it. For `__builtin_va_list` there is already a macro in the CDT parser, the same holds for `__builtin_va_arg` (but this is unfortunately not correct).